src/frontend/apps/lti_site/components/SelectContent/utils.ts

```diff
@@ -0,0 +1,84 @@
+import { defineMessages } from 'react-intl';
+
+import { Nullable } from 'lib-common';
+
+interface ContentItemsStructure {
+  '@context': string;
+  '@graph': {
+    '@type': string;
+    url: string;
+    title?: Nullable<string>;
+    text?: Nullable<string>;
+    frame: [];
+  }[];
+}
+
+export const buildContentItems = (
+  ltiUrl: string,
+  title: Nullable<string>,
+  description: Nullable<string>,
+  ltiSelectFormData: {
+    [key: string]: string;
+  },
+  setContentItemsValue: (value: string) => void,
+) => {
+  const contentItems: ContentItemsStructure = {
+    '@context': 'http://purl.imsglobal.org/ctx/lti/v1/ContentItem',
+    '@graph': [
+      {
+        '@type': 'ContentItem',
+        url: ltiUrl,
+        frame: [],
+      },
+    ],
+  };
+
+  if (title) {
+    contentItems['@graph'][0].title = title;
+  }
+  if (description) {
+    contentItems['@graph'][0].text = description;
+  }
+
+  if (ltiSelectFormData?.activity_title) {
+    contentItems['@graph'][0].title = ltiSelectFormData?.activity_title;
+  }
+  if (ltiSelectFormData?.activity_description) {
+    contentItems['@graph'][0].text = ltiSelectFormData?.activity_description;
+  }
+
+  setContentItemsValue(JSON.stringify(contentItems));
+};
+
+export const commonMessages = defineMessages({
+  addVideo: {
+    defaultMessage: 'Add a video',
+    description: `Text displayed on a button to add a new video.`,
+    id: 'components.SelectContent.commonMessages.addVideo',
```

Comment:
```suggestion
    id: 'components.SelectContent.utils.addVideo',
```
No ?
If so, the other ids need to be changed too.

Reply:
I choose `commonMessages` because this is what they are. Should I create a module `commonMessages` instead ?

Reply:
As you want, but she should keep it consistent IMO, see https://github.com/openfun/marsha/blob/ba134dc98b115d11490fd2978d6ddb3b9cc06452/src/frontend/apps/lti_site/apps/classroom/components/SelectContent/utils.ts

Reply:
From a crowdin perspective, i think we use the path as an id instead of the exported variable name.

Reply:
In that case I prefer to keep `commonMessages`

Reply:
:+1: